packages/core-data/src/hooks/use-entity-records.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Internal dependencies
+ */
+import useQuerySelect from './use-query-select';
+import { store as coreStore } from '../';
+import { Status } from './constants';
+
+interface EntityRecordsResolution< RecordType > {
+	/** The requested entity record */
+	records: RecordType[] | null;
+
+	/**
+	 * Is the record still being resolved?
+	 */
+	isResolving: boolean;
+
+	/**
+	 * Is the record resolved by now?
+	 */
+	hasResolved: boolean;
+
+	/** Resolution status */
+	status: Status;
+}
+
+/**
+ * Resolves the specified entity records.
+ *
+ * @param  kind      Kind of the requested entities.
+ * @param  name      Name of the requested entities.
+ * @param  httpQuery HTTP query for the requested entities.
```

Comment:
Nit: it's `query` in other places:
https://github.com/WordPress/gutenberg/blob/f289d911e50de9438b67512c112cd1fcd2e10691/packages/core-data/src/resolvers.js#L131

That could be `queryArgs` as well because they are included in the URL:

https://github.com/WordPress/gutenberg/blob/f289d911e50de9438b67512c112cd1fcd2e10691/packages/core-data/src/resolvers.js#L162-L165

They become `args` which is too general in this context:
https://github.com/WordPress/gutenberg/blob/f289d911e50de9438b67512c112cd1fcd2e10691/packages/url/src/add-query-args.js#L23

Reply:
Good note – I just renamed it to `queryArgs` to avoid collision with the `query` from `useQuerySelect`